### ISSUE TITLE
AWSMobileClient, updateUserAttributes completionHandler not called when success

### ISSUE THREAD

**Issue by JesusMartinAlonso** [aws-amplify/aws-sdk-ios#1162]
**Describe the bug**

I need to update the user attributes of an user. I am using  
`updateUserAttributes(attributeMap: [String: String], completionHandler: @escaping (([UserCodeDeliveryDetails]?, Error?) -> Void))` of `AWSMobileClient` 

The attributes are updated successfully but the `completionHandler` is not getting called. I have looked the implementation of the function and there is not a call to the `ompletionHandler` closure when the attributes are updated successfully.

```
public func updateUserAttributes(attributeMap: [String: String], completionHandler: @escaping (([UserCodeDeliveryDetails]?, Error?) -> Void)) {
        let attributes = attributeMap.map {AWSCognitoIdentityUserAttributeType.init(name: $0, value: $1) }
        self.userpoolOpsHelper.currentActiveUser!.update(attributes).continueWith { (task) -> Any? in
            if let error = task.error {
                completionHandler(nil, self.getMobileError(for: error))
            } else if let result = task.result {
                var codeDeliveryDetailsList: [UserCodeDeliveryDetails] = []
                if result.codeDeliveryDetailsList != nil {
                    for codeDeliveryDetail in result.codeDeliveryDetailsList! {
                        codeDeliveryDetailsList.append(self.getUserCodeDeliveryDetails(codeDeliveryDetail))
                    }
                }
            }
            return nil
        }
    }
```

**To Reproduce**
1.- Install MobileClient
2.- Do login with an existing user in a user pool
3.- Try to update userAttributes using AWSMobileClient:
```AWSMobileClient.sharedInstance().updateUserAttributes(attributeMap: userAttributes)```

**Which AWS service(s) are affected?**

AWSMobileClient

**Expected behavior**

**Environment(please complete the following information):**
 - SDK Version: 2.8.2
 - Dependency Manager: Cocoapods
 - Swift Version : 4.2

**Device Information (please complete the following information):**
 - Device: Iphone XS
 - iOS Version: 12.1
 - Specific to simulators:

**Comment by minbi** [comment]
Hi @JesusMartinAlonso ,

Thanks for reporting this issue and looking at the code. We will investigate the missing callback and post back here. For issues like this you may also open a pull request, but we'll take action on this one.

**Comment by rohandubal** [comment]
Hello @JesusMartinAlonso 

This issue is resolved in SDK version `2.8.3`. Please let us know if you continue facing the issues.

Thanks,
Rohan
